src/lib.rs

```diff
@@ -349,6 +354,32 @@ pub trait CongAlg<I: Ipc> {
     /// Create a new instance of the CongAlg to manage a new flow.
     /// Optionally copy any configuration parameters from `&self`.
     fn new_flow(&self, control: Datapath<I>, info: DatapathInfo) -> Self::Flow;
+
+}
+
+/// Structs implementing [`portus::CongAlg`](./trait.CongAlg.html) must also implement this trait
+/// (and must be annotated with [`portus_export::register_ccp_alg`]()) 
+/// in order to expose their algorithm for use in other programs, such as the CCP command-line utility.
+///
+/// The expected use of this trait in a calling program is as follows:
+/// ```
+/// let args = CongAlgBuilder::register();
+/// let matches = app.get_matches_from(std::env::args_os());
+/// let alg = CongAlgBuilder::with_arg_matches(matches);
+/// ```
```

Comment:
does this doctest compile?

Reply:
hm, it does but to be honest i feel like it shouldn't have lol

Reply:
added no-run